### PR TITLE
Fix compiler warnings related to fields.

### PR DESCRIPTION
**Pull request by samelhusseini**
## The basics
- [x] I branched from develop
- [x] My pull request is against develop
- [x] My code follows the [style guide](https://developers.google.com/blockly/guides/modify/web/style-guide)

### Resolves

Fixes 34 warnings related to fields.

### Proposed Changes

``bumpNeighbours`` is @package.
Fix unregister calls to delete from map rather than set to undefined.
Make ``SERIALIZABLE`` and ``EDITABLE`` not constants.
Fix ``doValueUpdate_`` parameter to be consistent with super class. 

### Reason for Changes

Compiler warnings.

### Test Coverage

Tested on:

### CODE REVIEW

File: core/block.js [hunk 1]
```diff
@@ -1106,7 +1107,7 @@ Blockly.Block.prototype.renameVarById = function(oldId, newId) {
 Blockly.Block.prototype.getFieldValue = function(name) {
   var field = this.getField(name);
   if (field) {
-    return field.getValue();
+    return String(field.getValue());
```
Comment (samelhusseini) [inline core/block.js]: This is an interesting one. field.getValue() can be of any type internally, but getFieldValue promises a string. Casting to string might not be desired, we could instead return a ``{*}``. Thoughts @rachel-fenichel?

Comment (BeksOmega) [reply core/block.js]: I think it needs to return `{*}` because `String()` will break object values.

Comment (samelhusseini) [reply core/block.js]: Yep, that’s a fair point. I’ll change it to return {string} instead

File: core/field_colour.js [hunk 2]
```diff
@@ -196,7 +195,7 @@ Blockly.FieldColour.prototype.doClassValidation_ = function(opt_newValue) {
 
 /**
  * Update the value of this colour field, and update the displayed colour.
- * @param {string} newValue The new colour in '#rrggbb' format.
+ * @param {*} newValue The value to be saved. A colour in '#rrggbb' format.
```
Comment (rachel-fenichel) [inline core/field_colour.js]: I think these should say something like "the default validator guarantees that this is a colour in '#rrggbb' format", and similar for the other `doValueUpdate_` functions.

File: core/field_variable.js [hunk 3]
```diff
@@ -144,7 +143,8 @@ Blockly.FieldVariable.prototype.initModel = function() {
     return; // Initialization already happened.
   }
   var variable = Blockly.Variables.getOrCreateVariablePackage(
-      this.workspace_, null, this.defaultVariableName, this.defaultType_);
+      this.sourceBlock_.workspace, null,
```
Comment (rachel-fenichel) [inline core/field_variable.js]: Is this ever a behaviour change?

Comment (rachel-fenichel) [reply core/field_variable.js]: Ah never mind, I see.

File: core/field_colour.js (hunk 2 above)
Comment (samelhusseini) [reply core/field_colour.js]: Done.

**Comment by NeilFraser** [comment]
This appears to break the compiler: #3152

**Pull request merged**